runtime/cudaq/builder/kernel_builder.cpp

```diff
@@ -510,50 +463,45 @@ QuakeValue qalloc(ImplicitLocOpBuilder &builder, QuakeValue &size) {
   return QuakeValue(builder, qubits);
 }
 
-QuakeValue qalloc(ImplicitLocOpBuilder &builder, std::size_t hash,
+QuakeValue qalloc(ImplicitLocOpBuilder &builder, const double *dataPtr,
                   std::size_t size) {
   auto *context = builder.getContext();
   auto parentModule =
       builder.getBlock()->getParentOp()->getParentOfType<ModuleOp>();
 
-  // Add the global for the state data to the module
-  if (failed(parseSourceString(getStateDataGlobalIntrinsic(hash, size),
-                               parentModule.getBody(),
-                               ParserConfig{context, false})))
-    throw std::runtime_error("Could not create code for state global data.");
-
-  // Add the function allowing one to set the state vector data to the module
-  if (failed(parseSourceString(getSetStateInstrinsic(hash),
-                               parentModule.getBody(),
-                               ParserConfig{context, false})))
-    throw std::runtime_error(
-        "Could not create code for setting the state global data.");
+  // Add a unique global for the state data to the module. The variable will be
+  // in the .bss section. invokeCode() will set this variable to point to the
+  // initializer data.
+  auto complexTy = ComplexType::get(builder.getF64Type());
+  auto stateDataTy = cudaq::cc::PointerType::get(
+      cudaq::cc::ArrayType::get(context, complexTy, size));
+  cudaq::cc::GlobalOp globalData;
+  {
+    OpBuilder::InsertionGuard guard(builder);
+    static std::size_t counter = 0;
+    builder.setInsertionPointToEnd(parentModule.getBody());
+    auto backingData = builder.getF64ArrayAttr(ArrayRef<double>{*dataPtr});
```

Comment:
If I understand this correctly - the concern I have here is that we would be backing the actual state data into the ModuleOp as part of this GlobalOp. What if we have a 30 qubit problem, this will store all 2**30 complex values in the Module, right?

The current implementation was trying to get around the need to back the values into the module op. The data exists in user-space so the idea was to just keep it there and only set a pointer to the data in the Module / Global when it is needed (when the code is invoked)